docs/translations.rst

```diff
@@ -6,6 +6,38 @@ localization strings. For more comprehensive information, consult the
 `Django Docs translations page <https://docs.djangoproject.com/en/dev/topics/i18n/translation/>`_
 or `this helpful blog post <http://blog.bessas.me/post/65775299341/using-gettext-in-django>`_.
 
+How translations work
+---------------------
+There are three major components to the translations system - identifying
+strings that need to be translated, having human translators write translations
+for those strings, and then actually inserting those translations when
+appropriate.
+
+1. ``./manage.py makemessages`` goes over the source code and pulls out strings
+   that have been tagged for translation. It does not actually execute code, so
+   you really can't do anything fancy in these strings.
+2. Transifex is a third party site we use for translating these strings. Human
+   translators look at the strings generated by makemessages and type up
+   appropriate translations. They have no context beyond what's in the string
+   itself. Be nice to them and give template variables names, so instead of "{}
+   version" they see "{build_date} version", which will be easier to understand.
+3. Finally, strings are actually translated when the code is executed on the
+   production servers. ``gettext`` is a simple function call that takes the
+   string provided and looks for it in the translations for the active language.
+   If present, it returns the corresponding translation. You can think of it
+   like this
+
+   .. code-block:: python
+
+      TRANSLATIONS = {
+          "en": {"hello": None},
+          "es": {"hello": "hola"},
+          "fra": {"hello": "bonjour"},
+      }
+
+      def dumb_gettext(str):
+          return TRANSLATIONS[get_language()][str] or str
+
```

Comment:
What would you think of adding something like this? I comment on f strings in translations pretty often in PRs.

---

This means that the exact string passed to `gettext` must exist in the translation files. Therefore, f strings cannot be used with `gettext`. Code like this

```
gettext(f"Today is {day_of_week}")
```

will result in a translation file entry like this

```
#: corehq/apps/accounting/filters.py corehq/apps/accounting/forms.py
msgid "Today is {day_of_week}"
msgstr "Hoy es {day_of_week}"
```

but when code executes, the string passed to gettext will be something like "Today is Friday" which will not be found in translations.

Because of this, instead of using f strings with gettext, use `format` on the output of `gettext`: `gettext("Today is {}").format(day_of_week)`

Reply:
...I now see this PR originated in a discussion of f strings.

Reply:
Just added a table with some examples